dace/transformation/transformation.py

```diff
@@ -55,7 +84,11 @@ def expressions():
         raise NotImplementedError
 
     @staticmethod
-    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+    def can_be_applied(graph: Union[SDFG, SDFGState],
```

Comment:
`graph: Union[SDFG, SDFGState]` is confusing. Can we have separate functions with different names for SDFG and SDFGState? can we replace `graph` with `sdfg` and `sdfg_state` depending on the concrete type?

Reply:
Unfortunately not, because a transformation can be used on multiple states (and then graph is `SDFG`) or a single state. If you want, you can make a separate PR where Transformation is extended to `SinglestateTransformation` and `InterstateTransformation`.

Reply:
Yes, probably it should go in a separate PR